manual/server-examples/src/docs/dev/index.asciidoc

```diff
@@ -4,6 +4,18 @@ Extending the Neo4j Server
 
 The Neo4j Server can be extended by either plugins or unmanaged extensions.
 
+[WARNING]
+====
+When running your own code alongside Neo4j in the same JVM, there are a few things you should keep in mind:
+
+* Don't create or retain more objects than you strictly need to.
+  Large caches are particularly good at promoting many objects to the old generation, thus increasing the need for expensive full garbage collections.
+* Don't use internal Neo4j APIs.
+  They may change in future releases, and they might not always work as intended in the context of your code.
+* Don't use Java object serialization or reflection in your code, or in any runtime dependency you use, if you can avoid it.
+  If you can't, make sure that the `-XX:+TrustFinalNonStaticFields` JVM flag is disabled in `neo4j-wrapper.conf`.
```

Comment:
If you cannot avoid using Java object serialization and reflection, ensure that the....